openedx/core/djangoapps/credit/tasks.py

```diff
@@ -208,17 +204,24 @@ def _get_proctoring_requirements(course_key):
     # process
     from edx_proctoring.api import get_all_exams_for_course
 
-    requirements = [
-        {
-            'namespace': 'proctored_exam',
-            'name': exam['content_id'],
-            'display_name': exam['exam_name'],
-            'criteria': {},
-        }
-        for exam in get_all_exams_for_course(unicode(course_key))
-        # practice exams do not count towards eligibility
-        if exam['is_proctored'] and exam['is_active'] and not exam['is_practice_exam']
-    ]
+    requirements = []
+    for exam in get_all_exams_for_course(unicode(course_key)):
+        if exam['is_proctored'] and exam['is_active'] and not exam['is_practice_exam']:
+            try:
+                usage_key = UsageKey.from_string(exam['content_id'])
+                proctor_block = modulestore().get_item(usage_key)
+            except (InvalidKeyError, ItemNotFoundError):
+                LOGGER.info("Invalid content_id '%s' for proctored block '%s'", exam['content_id'], exam['exam_name'])
+                proctor_block = None
+
+            requirements.append(
+                {
+                    'namespace': 'proctored_exam',
+                    'name': exam['content_id'],
+                    'display_name': exam['exam_name'],
+                    'start_date': proctor_block.start if proctor_block and proctor_block.start else None,
+                    'criteria': {},
+                })
```

Comment:
Does it make sense to add the proctoring requirement if we can't find the proctored exam in the courseware?